server/sonar-process-monitor/src/main/java/org/sonar/application/process/CommandFactoryImpl.java

```diff
@@ -44,20 +47,50 @@
 
   private final AppSettings settings;
 
-  public JavaCommandFactoryImpl(AppSettings settings) {
+  public CommandFactoryImpl(AppSettings settings) {
     this.settings = settings;
   }
 
   @Override
-  public JavaCommand createEsCommand() {
-    File homeDir = settings.getProps().nonNullValueAsFile(ProcessProperties.PATH_HOME);
-    return newJavaCommand(ProcessId.ELASTICSEARCH, homeDir)
-      .addJavaOptions("-Djava.awt.headless=true")
-      .addJavaOptions(settings.getProps().nonNullValue(ProcessProperties.SEARCH_JAVA_OPTS))
-      .addJavaOptions(settings.getProps().nonNullValue(ProcessProperties.SEARCH_JAVA_ADDITIONAL_OPTS))
-      .setClassName("org.sonar.search.SearchServer")
-      .addClasspath("./lib/common/*")
-      .addClasspath("./lib/search/*");
+  public EsCommand createEsCommand(AppSettings settings) {
+    File homeDir = this.settings.getProps().nonNullValueAsFile(ProcessProperties.PATH_HOME);
+    File executable = new File(homeDir, getExecutable());
+    if (!executable.exists()) {
+      throw new IllegalStateException("Cannot find elasticsearch binary");
+    }
+
+    Map<String, String> settingsMap = new EsSettings(this.settings.getProps()).build();
+
+    EsCommand res = new EsCommand(ProcessId.ELASTICSEARCH)
+      .setWorkDir(executable.getParentFile().getParentFile())
+      .setExecutable(executable)
+      .setArguments(this.settings.getProps().rawProperties())
+      .setClusterName(settingsMap.get("cluster.name"))
+      .setHost(settingsMap.get("network.host"))
+      // TODO add argument to specify log4j configuration file
+      // TODO add argument to specify yaml configuration file
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135)